rfcs/text/0008-sip-fields.md

```diff
@@ -0,0 +1,242 @@
+-->
+
+## Scope of impact
+
+No impact expected as SIP fieldsets are new, and will not impact any existing fields.
+<!--
+Stage 2: Identifies scope of impact of changes. Are breaking changes required? Should deprecation strategies be adopted? Will significant refactoring be involved? Break the impact down into:
+ * Ingestion mechanisms (e.g. beats/logstash)
+ * Usage mechanisms (e.g. Kibana applications, detections)
+ * ECS project (e.g. docs, tooling)
+The goal here is to research and understand the impact of these changes on users in the community and development teams across Elastic. 2-5 sentences each.
+-->
+
+## Concerns
+
+Normalization, and the degree of normalization, of SIP URI fields may be an issue for discussion based on the potential implementation of ingesting SIP call records for the purposes of review for e.g. various types of communications fraud (e.g. should PSTN numbers be normalized with international dial codes, should implementaiton include capabilities to define internal call plans for more effective analysis, etc.)
+
+Normalization of SIP/SDP and real time communication protocol connections may require the definition of a field similar to network.community_id to allow for the tracking of the full scope of a connection. Additionally the initial SDP setup phase often includes
+multiple audio/video codec definitions which may be difficult to normalize in such a way as to make analysis of the call setup phase effective.
+
+Utilizing SIP fields in combination with network performance indicators (IP SLA, QOS settings, jitter, mos, etc.) would also be of interest to many users looking at SIP logging.
+<!--
+Stage 1: Identify potential concerns, implementation challenges, or complexity. Spend some time on this. Play devil's advocate. Try to identify the sort of non-obvious challenges that tend to surface later. The goal here is to surface risks early, allow everyone the time to work through them, and ultimately document resolution for posterity's sake.
+-->
+
+<!--
+Stage 2: Document new concerns or resolutions to previously listed concerns. It's not critical that all concerns have resolutions at this point, but it would be helpful if resolutions were taking shape for the most significant concerns.
+-->
+
+<!--
+Stage 3: Document resolutions for all existing concerns. Any new concerns should be documented along with their resolution. The goal here is to eliminate the risk of churn and instability by resolving outstanding concerns.
+-->
+
+<!--
+Stage 4: Document any new concerns and their resolution. The goal here is to eliminate risk of churn and instability by ensuring all concerns have been addressed.
+-->
+
+## Real-world implementations
+
+Packetbeat Implementation (packet/protocol analysis)
+Zeek Implementation (packet/protocol/connection analysis)
+Cisco Call Manager (Log ingestion)
+Microsoft Lync (Log Analysis)
```

Comment:
Here we're looking to list the real world implementation of using the SIP ECS fields. You can keep Packetbeat here, since the goal is for it to be adjusted to match the ECS fields.

The other ones should be removed for now, though. They're already correctly mentioned in the "data sources" section.

Note that when the Zeek SIP parser is adjusted to follow ECS we can list it here as well. But hopefully it won't be adjusted until this RFC reaches at least stage 2, when fields will be more stable.

Reply:
commented out